lib/src/types/connector.ts

```diff
@@ -0,0 +1,61 @@
+export type ConnectorType = 'PERSONAL_BANK' | 'BUSINESS_BANK' | 'INVESTMENT'
+
+export type CredentialSelectOption = {
+  /** Value of the option */
+  value: string
+  /** Displayable text or label of the option */
+  label: string
+}
+
+export type ConnectorCredential = {
+  /** parameter label that describes it */
+  label: string
+  /** parameter key name */
+  name: string
+  /** type of parameter to create the form */
+  type?: 'text' | 'password' | 'number' | 'image' | 'select'
```

Comment:
FYI, in `pluggy-js` I've defined this field as `type?: CredentialType`, defined like this:

```ts
export const CREDENTIAL_TYPES = ['number', 'password', 'text', 'image', 'select'] as const
/**
 * @typedef CredentialType
 * credential type, used to show a proper form input to the user
 *  'number' -> numeric only data
 *  'text' -> alpha-numeric data
 *  'password' -> alpha-numeric password, must be obfuscated
 *  'image' -> a QR code needs to be decoded (QR is provided in the credential.data field)
 *  'select' -> credential has to be picked from values listed in credential.options field
 */
export type CredentialType = typeof CREDENTIAL_TYPES[number]
```

Reply:
Updated.

Reply:
@Gabrielpanga see below a few examples, in this case from the [connect webapp](https://github.com/pluggyai/connect/blob/9e72718e921708b5ef6c83463af70ade7402a88e/src/utils/appWrapper.ts#L75-L109), of a benefit a client could get by having the ability to choose from both available `CredentialType` union type, or `CREDENTIAL_TYPES` exhaustive list of possible values.

In this case I use `CONNECTOR_TYPES` and `COUNTRY_CODES` lists to properly validate app string input params. 
If I hadn't thouse, then I would have to create my own list mimicking the respective `ConnectorType` and `CountryCode` union types. The issue there is that they could potentially get out of date if we ever added/updated/removed possible values from those types. By providing the exhaustive list as well, that issue is mitigated by only requiring the client app to update to the latest version and that would be it.

```ts
/**
 * Extract valid ConnectorTypes from array-like string parameter, separated by comma.
 */
const parseConnectorTypes = (
  connectorTypesParam: string | null
): ConnectorType[] | undefined => {
  if (!connectorTypesParam) {
    return undefined
  }

  return connectorTypesParam
    .replace(/ /g, '')
    .split(',')
    .filter((connectorTypeString): connectorTypeString is ConnectorType =>
      CONNECTOR_TYPES.includes(connectorTypeString as ConnectorType)
    )
}

/**
 * Extract valid CountryCodes list from array-like string parameter, separated by comma.
 */
const parseCountries = (
  countriesParam: string | null
): CountryCode[] | undefined => {
  if (!countriesParam) {
    return undefined
  }

  return countriesParam
    .replace(/ /g, '')
    .split(',')
    .filter((countryCode): countryCode is CountryCode =>
      COUNTRY_CODES.includes(countryCode as CountryCode)
    )
}

```